packages/fuels-core/src/types/transaction_builders/blob.rs

```diff
@@ -0,0 +1,205 @@
+use std::{fmt::Debug, iter::repeat};
+
+use async_trait::async_trait;
+use fuel_crypto::Signature;
+use fuel_tx::{
+    field::{Policies as PoliciesField, Witnesses},
+    policies::{Policies, PolicyType},
+    BlobIdExt, Chargeable, Output, Transaction as FuelTransaction, UniqueIdentifier, Witness,
+};
+use fuel_types::bytes::padded_len_usize;
+use itertools::Itertools;
+
+use super::{
+    generate_missing_witnesses, impl_tx_builder_trait, resolve_fuel_inputs, BuildableTransaction,
+    Strategy, TransactionBuilder, UnresolvedWitnessIndexes,
+};
+use crate::{
+    constants::SIGNATURE_WITNESS_SIZE,
+    traits::Signer,
+    types::{
+        errors::{error, error_transaction, Result},
+        input::Input,
+        transaction::{BlobTransaction, EstimablePredicates, Transaction, TxPolicies},
+        DryRunner,
+    },
+    utils::{calculate_witnesses_size, sealed},
+};
+
+#[derive(Default, Clone, Debug, PartialEq)]
+pub struct Blob {
+    pub data: Vec<u8>,
```

Comment:
```suggestion
    data: Vec<u8>,
```

Reply:
Initially wanted to make it public but you're right. We're better of hiding the field. Could add caching later on for the id if needed.